src/core/RenderManager.cpp

```diff
@@ -88,30 +88,7 @@ void RenderManager::renderNextTrack()
 		// for multi-render, prefix each output file with a different number
 		int trackNum = m_tracksToRender.size() + 1;
 
-		// create a renderer for this track
-		m_activeRenderer = make_unique<ProjectRenderer>(
-				m_qualitySettings,
-				m_outputSettings,
-				m_format,
-				pathForTrack(renderTrack, trackNum));
-
-		if ( m_activeRenderer->isReady() )
-		{
-			// pass progress signals through
-			connect( m_activeRenderer.get(), SIGNAL( progressChanged( int ) ),
-					this, SIGNAL( progressChanged( int ) ) );
-
-			// when it is finished, render the next track
-			connect( m_activeRenderer.get(), SIGNAL( finished() ),
-					this, SLOT( renderNextTrack() ) );
-
-			m_activeRenderer->startProcessing();
-		}
-		else
-		{
-			qDebug( "Renderer failed to acquire a file device!" );
-			renderNextTrack();
-		}
+		render( pathForTrack(renderTrack, trackNum) );
```

Comment:
Previously if one track failed to render (`m_activeRenderer->isReady()` returns false), we would still call `renderNextTrack()` to continue rendering the remaining tracks. The new behavior is that once one track returns false in `m_activeRenderer->isReady()`, we end rendering completely and `emit finished()`.

I don't know when `isReady()` is expected to return false - should we worry about preserving the previous behavior?

Reply:
Can we replace `emit finished()` with `renderNextTrack()`?

Reply:
@PhysSong I think that should work. Done via daee35ebe86bf91c4c4ba40b1600e5f927ae1d08.